membership_handling.py

```diff
@@ -484,8 +484,11 @@ async def set_membership(self, interaction, member_id, date, manual=True, actor=
         return True
 
     async def del_membership(self, interaction, member_id: int, text, dm_flag=True, manual=True, vtuber=None):
-        guild_id = interaction.guild_id
-        author_id = interaction.user.id
+        guild_id = interaction.guild.id
+        if isinstance(interaction, discord.Message):
+            author_id = interaction.author.id
+        else:
+            author_id = interaction.user.id
```

Comment:
Is there any reason you want to differentiate between interaction and message? In my opinion there is really no reason to do so anymore because the bot cannot even process normal messages anymore which could lead to this code called from a $delMember instead of the slash command

Reply:
If you reject a membership from the logging channel, the "interaction" passed to the del_membership function is actually a Message object, and would fail without this check.